src/server/game/Handlers/MiscHandler.cpp

```diff
@@ -1117,36 +1117,36 @@ void WorldSession::HandleInspectHonorStatsOpcode(WorldPacket& recv_data)
     SendPacket(&data);
 }
 
-void WorldSession::HandleWorldTeleportOpcode(WorldPacket& recv_data)
+void WorldSession::HandleWorldTeleport(WorldPacket& message)
 {
-    uint32 time;
-    uint32 mapid;
-    float PositionX;
-    float PositionY;
-    float PositionZ;
-    float Orientation;
+    Position    position;
+    uint32      continentID;
+    uint32      requestTimeMs;
+    uint64      player;
 
-    recv_data >> time;                                      // time in m.sec.
-    recv_data >> mapid;
-    recv_data >> PositionX;
-    recv_data >> PositionY;
-    recv_data >> PositionZ;
-    recv_data >> Orientation;                               // o (3.141593 = 180 degrees)
+    if (!AccountMgr::IsGMAccount(GetSecurity()))
+    {
+        SendNotification(LANG_YOU_NOT_HAVE_PERMISSION);
+        return;
+    }
 
-    LOG_DEBUG("network", "WORLD: Received CMSG_WORLD_TELEPORT");
+    message >> requestTimeMs;
+    message >> continentID;
+    message >> player;
+    message >> position.m_positionX;
+    message >> position.m_positionY;
+    message >> position.m_positionZ;
+    message >> position.m_orientation;
 
-    if (GetPlayer()->IsInFlight())
+    Player* playerPtr = player ? ObjectAccessor::FindPlayer((ObjectGuid)player) : GetPlayer();
+
+    if (!playerPtr)
     {
-        LOG_DEBUG("network", "Player '{}' ({}) in flight, ignore worldport command.", GetPlayer()->GetName(), GetPlayer()->GetGUID().ToString());
+        SendNotification("Player not found");
         return;
     }
 
-    LOG_DEBUG("network", "CMSG_WORLD_TELEPORT: Player = {}, Time = {}, map = {}, x = {}, y = {}, z = {}, o = {}", GetPlayer()->GetName(), time, mapid, PositionX, PositionY, PositionZ, Orientation);
```

Comment:
dont get rid of this, debugs in opcodes are good

Reply:
I don't see the point but I don't care, so it's done 😂